### PR TITLE
add hyp link

### PR DESCRIPTION
**Pull request by jairad26**

### CODE REVIEW

File: src/commands/link/index.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    // get list of the projects for the user in this org, if any have no repoId, ask if they want to link it, or give option of none.
+    // If they pick an option, connect repo. If none, ask if they want to create a new project, prompt for name, and connect repoId to project
+    const projects = await getProjectsByOrgGraphQLRequest(res.jwt, res.orgId)
+
+    const projectsNoRepoId = projects.filter(project => !project.repoId)
+
+    let selectedProject = null
+
+    if (projectsNoRepoId.length > 0) {
```
Comment (kristencheung) [inline src/commands/link/index.ts:72]: I think we want to give them the option to choose between creating a new project or mapping to an existing project, as opposed to prompting first to map an existing project

File: src/commands/link/index.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  ]
+
+  static override flags = {}
+
+  public async run(): Promise<void> {
+    // check if the directory has a .git/config with a remote named 'origin', if not, throw an error and ask them to set that up
+    const gitConfigFilePath = getGitConfigFilePath()
+
+    if (!fileExists(gitConfigFilePath)) {
+      throw new Error('No .git/config file found, please set up a remote git repository')
```
Comment (kristencheung) [inline src/commands/link/index.ts:24]: I think just "No remote git repository found" is fine

File: src/commands/link/index.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      // github app installation flow here
+    } else  {
+      installationId = res.installationIds[gitOwner]
+    }
+
+    // call hypermode getRepoId with the installationId and the git url, if it returns a repoId, continue, if not, throw an error
+    const repoId = await sendGetRepoIdGraphQLRequest(res.jwt, installationId, gitUrl)
+
+    if (!repoId) {
+      throw new Error('No repoId found for the given installationId and gitUrl')
```
Comment (kristencheung) [inline src/commands/link/index.ts:61]: are we showing this error message show to the user?

Comment (jairad26) [reply src/commands/link/index.ts:61]: yeah, basically like we couldnt fetch the repoId, so something went wrong

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**